source/extensions/filters/http/dynamic_forward_proxy/proxy_filter.cc

```diff
@@ -165,22 +173,13 @@ Http::FilterHeadersStatus ProxyFilter::decodeHeaders(Http::RequestHeaderMap& hea
   NOT_REACHED_GCOVR_EXCL_LINE;
 }
 
-void ProxyFilter::addHostAddressToFilterState(
-    const Network::Address::InstanceConstSharedPtr& address) {
+void ProxyFilter::addHostAddressToFilterState(const Network::Address::InstanceConstSharedPtr& address) {
+  ASSERT(address);  // null pointer checks must be done before calling this function.
```

Comment:
I was confused for a minute about how a "const T&" could even be null.... then I realized that T is "SharedPtr". Carry on! *facepalm*